pushconfig.sh

```diff
@@ -434,23 +454,27 @@ function update_configuration() {
 }
 
 api_port=0
-if [ ! $APIGATEWAY_URL ]; then
-    # Start a kubectl proxy in the background, to access the services API
-    kubectl proxy -p 0 >proxy.port &
-    pid="$!"
-    sleep 3
-    kill -0 "${pid}" # Check that the proxy is started (the command "kill -0" returns an error status when the pid does not exist)
-
-    api_port=$(cat proxy.port | cut -d':' -f 2)
-    echo "API Proxy started on port $api_port (pid $pid)"
-
-    # Make sure to stop the proxy when this script ends (this function will be executed only at EXIT)
-    function cleanup() {
-        echo "Stopping proxy"
-        kill -9 "$pid" || true
-        rm proxy.port
-    }
-    trap cleanup EXIT
+if $NO_DRY_RUN; then
+    if [ ! $APIGATEWAY_URL ]; then
+        # Start a kubectl proxy in the background, to access the services API
+        kubectl proxy -p 0 >proxy.port &
+        pid="$!"
+        sleep 3
+        kill -0 "${pid}" # Check that the proxy is started (the command "kill -0" returns an error status when the pid does not exist)
+
+        api_port=$(cat proxy.port | cut -d':' -f 2)
+        echo "API Proxy started on port $api_port (pid $pid)"
+
+        # Make sure to stop the proxy when this script ends (this function will be executed only at EXIT)
+        function cleanup() {
+            echo "Stopping proxy"
+            kill -9 "$pid" || true
+            rm proxy.port
+        }
+        trap cleanup EXIT
+    fi
+else
+    echo "[DRY RUN] Do not start kubectl proxy"
```

Comment:
Why ?
The dry run should still be able to connect to the cluster (to know if there are any changes) ?